### ISSUE TITLE
Payform Hanging on Processing

### ISSUE THREAD

**Issue by jperanson** [Beanstream/beanstream-payform#20]
### Expected behaviour:

Once valid credit card information is submitted, the payform should return a response

### Actual behaviour:

The "processing" icon stays on screen indefinitely - for up to several hours with no transaction reaching beanstreams servers. The last lines in processing in console are:

beanstream_payform.js:496
beanstream_payfields.js:1594

### Steps to reproduce:

Submit valid credit card information, select Pay button 

### Browsers:

Safari 9.3.2

**Comment by bowens-bambora** [comment]
Do you see anything in the console logs? I cannot reproduce the issue, but we will continue to look into the matter.

**Comment by jperanson** [comment]
No. Console shows no errors.

The last lines of execution show:

> Communication verified. beanstream_payform.js:496
> Starting Beanstream Payfields... beanstream_payfields.js:1594


The payform is submitted, the console just hangs there. The user waits for
up to an hour and never sees what is usually the next step:
** FormController sending message beanstream_toknizationForm_complete :
{"cardInfo":{"code":"...

This is happening to about 10% of users. Trying to confirm if it is brower
specific.

**Comment by jperanson** [comment]
I will also note that when the payform popped up, it was left justified on the customer's screen, as opposed to centered on the page as it typically displays.

**Comment by jperanson** [comment]
Issue also presented in Google Chrome on a PC for the same user - "processing" screen showed indefinitely (30 minutes) and did not time out. The user then tried Microsoft edge and transaction completed successfully..

**Comment by aengusbates** [comment]
Fix applied and deployed.

**Comment by aengusbates** [comment]
Hi Jonah,

Have you been able to reproduce this issue in the past five days?

We have set up a New Relic job to monitor the Tokenization API and are getting a consistent return time of around 1s across North America. Payform will show the spinner panel for a minimum or 3s. I have added a timeout of 10s for Payfields, so the maximum wait on a return from either Payform or Payfields is now 10s. We have added a timeout error to Payfields and are now returning it as part of the Payform return object. 

I have also resolved a hand specific to Safari on certain versions of iOS.

As the fixes applied resolve all hang-like behaviour reproducible on our end, I'm inclined to close this bug. We can reopen it down the line if needed.

Cheeers, Aengus
